scene/2d/path_2d.cpp

```diff
@@ -30,13 +30,137 @@
 
 #include "path_2d.h"
 
+#include "core/config/project_settings.h"
 #include "core/math/geometry_2d.h"
 #include "scene/main/timer.h"
 
 #ifdef TOOLS_ENABLED
 #include "editor/themes/editor_scale.h"
 #endif
 
+bool PathDebug2D::debug_enabled = true;
+Color PathDebug2D::debug_paths_color = Color(0.1, 1.0, 0.7, 0.4);
+float PathDebug2D::debug_paths_width = 2.0;
+float PathDebug2D::debug_paths_sample_interval = 5.0;
+bool PathDebug2D::debug_paths_fish_bones_enabled = true;
+int PathDebug2D::debug_paths_fish_bones_interval = 4;
+
+Mutex PathDebug2D::update_callbacks_mutex;
+HashMap<Path2D *, Callable> PathDebug2D::update_callbacks;
+
+void PathDebug2D::add_update_callback(Path2D *p_path, Callable p_callback) {
+	MutexLock lock(update_callbacks_mutex);
+	update_callbacks.insert(p_path, p_callback);
+}
+
+void PathDebug2D::remove_update_callback(Path2D *p_path) {
+	MutexLock lock(update_callbacks_mutex);
+	update_callbacks.erase(p_path);
+}
+
+void PathDebug2D::emit_update_callbacks() {
+	MutexLock lock(update_callbacks_mutex);
+	for (KeyValue<Path2D *, Callable> &kv : update_callbacks) {
+		ERR_CONTINUE(kv.value.is_null());
```

Comment:
```suggestion
		ERR_CONTINUE(!kv.value.is_valid());
```
`Callable` should be checked against `is_valid` for checking if it can be called